python_modules/libraries/dagster-components/dagster_components/core/component_defs_builder.py

```diff
@@ -67,26 +67,31 @@ def load_components_from_context(context: ComponentLoadContext) -> Sequence[Comp
     raise NotImplementedError(f"Unknown component type {context.decl_node}")
 
 
+def find_local_component_types(component_path: Path) -> list[type[Component]]:
+    """Find all component types defined in a component directory."""
+    component_types = []
+    for py_file in component_path.glob("*.py"):
+        module_name = py_file.stem
+
+        module = load_module_from_path(module_name, component_path / f"{module_name}.py")
+
+        for _name, obj in inspect.getmembers(module, inspect.isclass):
+            assert isinstance(obj, type)
+            if is_registered_component_type(obj):
```

Comment:
Can't you use a local component type _without_ it being a registered component type? In general I thought we were not registering component local component types. If so then the right check is an `isinstance` check instead.

Reply:
Right now, the answer I believe is no, since this is the logic which determines if there's a matching local component type. Maybe the answer should be yes, though.